crates/driver/src/tests/cases/settle.rs

```diff
@@ -110,3 +113,141 @@ async fn high_gas_limit() {
         .unwrap();
     test.settle(&id).await.ok().await;
 }
+
+#[tokio::test]
+#[ignore]
+async fn discards_excess_settle_requests() {
+    let test = tests::setup()
+        .allow_multiple_solve_requests()
+        .pool(ab_pool())
+        .order(ab_order())
+        .solution(ab_solution())
+        .ethrpc_args(shared::ethrpc::Arguments {
+            ethrpc_max_batch_size: 10,
+            ethrpc_max_concurrent_requests: 10,
+            // This delay artificially slows down the settlement process, so the /settle calls are
+            // accumulated in the driver's queue.
+            ethrpc_batch_delay: std::time::Duration::from_secs(1),
+        })
+        .solve_deadline_timeout(chrono::Duration::seconds(4))
+        .done()
+        .await;
+
+    // MAX_SOLUTION_STORAGE = 5. Since this is hardcoded, no more solutions can be
+    // stored.
+    let solution_ids = join_all(vec![
+        test.solve(),
+        test.solve(),
+        test.solve(),
+        test.solve(),
+        test.solve(),
```

Comment:
Would be good to make `/solve` return errors when the queue is already full. Otherwise the solver could continue winning auctions and later get slashed because it can no longer submit them.

Reply:
Great idea, I will address it in the original PR then.